File: tkpdsession/src/main/java/com/tokopedia/session/addchangepassword/view/activity/AddPasswordActivity.java
package com.tokopedia.session.addchangepassword.view.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.fragment.app.Fragment;

import com.tokopedia.abstraction.base.view.activity.BaseSimpleActivity;
import com.tokopedia.session.addchangepassword.view.fragment.AddPasswordFragment;

/**
 * @author by yfsx on 23/03/18.
 *
 * For navigating to this class
 * {@link com.tokopedia.applink.internal.ApplinkConstInternalGlobal#ADD_PASSWORD}
 */

public class AddPasswordActivity extends BaseSimpleActivity {

    public static Intent newInstance(Context context) {
        Intent intent = new Intent(context,  AddPasswordActivity.class);
        intent.putExtras(new Bundle());
        return intent;
    }

    @Override
    protected Fragment getNewFragment() {
        return AddPasswordFragment.newInstance(new Bundle());
    }

}

File: features/user/managepassword/src/main/java/com/tokopedia/managepassword/addpassword/view/activity/AddPasswordActivity.kt
package com.tokopedia.managepassword.addpassword.view.activity

import androidx.fragment.app.Fragment
import com.tokopedia.abstraction.base.app.BaseMainApplication
import com.tokopedia.abstraction.base.view.activity.BaseSimpleActivity
import com.tokopedia.abstraction.common.di.component.HasComponent
import com.tokopedia.managepassword.addpassword.view.fragment.AddPasswordFragment
import com.tokopedia.managepassword.di.DaggerManagePasswordComponent
import com.tokopedia.managepassword.di.ManagePasswordComponent
import com.tokopedia.managepassword.di.module.ManagePasswordModule

/**
 * @author rival
 * @created 14/05/2020
 * @team : @minion-kevin
 *
 * For navigating to this class
 * @applink : [com.tokopedia.applink.internal.ApplinkConstInternalUserPlatform.ADD_PASSWORD]
 * @params
 *
 */

class AddPasswordActivity: BaseSimpleActivity(), HasComponent<ManagePasswordComponent> {

    override fun getScreenName(): String {
        return SCREEN_ADD_PASSWORD
    }

    override fun getComponent(): ManagePasswordComponent {
        return DaggerManagePasswordComponent.builder()
                .baseAppComponent((application as BaseMainApplication).baseAppComponent)
                .managePasswordModule(ManagePasswordModule(this))
                .build()
    }

    override fun getNewFragment(): Fragment {
        return AddPasswordFragment.createInstance()
    }

    companion object {
        private const val SCREEN_ADD_PASSWORD = "Add password page"
    }
}